ArduCopter/mode_zigzag.cpp

```diff
@@ -120,39 +138,53 @@ void ModeZigZag::save_or_move_to_destination(uint8_t dest_num)
             // A and B have been defined, move vehicle to destination A or B
             Vector3f next_dest;
             bool terr_alt;
-            if (calculate_next_dest(dest_num, stage == AUTO, next_dest, terr_alt)) {
+            if (calculate_next_dest(ab_dest, stage == AUTO, next_dest, terr_alt)) {
                 wp_nav->wp_and_spline_init();
                 if (wp_nav->set_wp_destination(next_dest, terr_alt)) {
                     stage = AUTO;
-#if SPRAYER_ENABLED == ENABLED
+                    ab_dest_stored = ab_dest;
                     // spray on while moving to A or B
-                    if (g2.zigzag_auto_pump_enabled) {
-                        copter.sprayer.run(true);
-                    }
-#endif
+                    spray(true);
                     reach_wp_time_ms = 0;
-                    if (dest_num == 0) {
-                        gcs().send_text(MAV_SEVERITY_INFO, "ZigZag: moving to A");
+                    if (auto_enabled == false || line_num == ZIGZAG_LINE_INFINITY) {
+                        gcs().send_text(MAV_SEVERITY_INFO, "ZigZag: moving to %s", (ab_dest == Destination::A) ? "A" : "B");
                     } else {
-                        gcs().send_text(MAV_SEVERITY_INFO, "ZigZag: moving to B");
+                        line_count++;
+                        auto_stage = AutoState::AB_MOVING;
+                        gcs().send_text(MAV_SEVERITY_INFO, "ZigZag: moving to %s (line %d/%d)", (ab_dest == Destination::A) ? "A" : "B", line_count, line_num);
                     }
                 }
             }
             break;
     }
 }
 
+void ModeZigZag::move_to_side()
+{
+    if (!dest_A.is_zero() && !dest_B.is_zero() && is_positive((dest_B - dest_A).length_squared())) {
```

Comment:
is_positive((dest_B - dest_A).length_squared()) will always be true as squared number is always positive

Reply:
Thank you. I changed to use !is_zero.